### PR TITLE
All products page

### PR DESCRIPTION
**Pull request by yanbutan**
## Changelog

- All products page search.
- Refactored full site search to use same logic as all products search.
- Added in missing category front matter tags.

## Note
- There are changes that are not mine, such as in apps/src/lib/index.js . Might have had been added here because i merged local to my branch locally.
- @candyyap-gvt will need your help to add in the target groups for the all of the products.

Just need to add to the frontmatter like this : 
Screenshot 2022-02-24 at 6 46 32 PM

### CODE REVIEW

File: apps/src/search/vue-search-app-full-page.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import Vue from "vue";
+import SearchFullSite from "./SearchFullSite.vue";
+import VueCompositionAPI from "@vue/composition-api";
+Vue.use(VueCompositionAPI);
+
+new Vue({
+  el: "#search-app-full-page",
```
Comment (teochengyong) [inline apps/src/search/vue-search-app-full-page.js:7]:
```suggestion
  el: "#search-app-full-site",
```
Shall we renamed this to full-site? I think this naming means exactly what it is doing

**Review by teochengyong — COMMENTED** [review]
Great use of composition api! Overall a great PR, still checking out the spacing.scss to see whether we can reuse existing css from the design system, for the typography scss not sure if it is possible to put it in the component itself instead of in the typography.scss. 

Let me spend some time to understand the composition apis and the lunr indexing abit more.

Can you check whether is possible to add simple tests to the lunr.js library? I think that's the major part that we should test since it contains some business logic.

A great refractor overall 👍

File: apps/src/search/vue-search-app-full-page.js (hunk 1 above)
Comment (yanbutan) [reply apps/src/search/vue-search-app-full-page.js:7]: ah okay yea I think should do that

**Comment by yanbutan** [comment]
> Great use of composition api! Overall a great PR, still checking out the spacing.scss to see whether we can reuse existing css from the design system, for the typography scss not sure if it is possible to put it in the component itself instead of in the typography.scss.
> 
> Let me spend some time to understand the composition apis and the lunr indexing abit more.
> 
> Can you check whether is possible to add simple tests to the lunr.js library? I think that's the major part that we should test since it contains some business logic.

Okay, I'll see if I can add some unit tests.

**Review by teochengyong — APPROVED** [review]
LGTM!

**Pull request merged**